### PR TITLE
Travis config + Fix release build by using an old version of the gradle android plugin

### PR DESCRIPTION
**Pull request by maxme**
New Travis config: build debug and release versions, also run lint checks.

There is a bug with NDK processing in 2.1+ https://code.google.com/p/android/issues/detail?id=223162 that prevents any release build, I had to downgrade to 2.0.0. Hopefully this will be fixed in 2.2.3+ or 2.3+.

### CODE REVIEW

File: Simplenote/gradle.properties-example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+simperiumAppId=FIXME
+simperiumAppKey=FIXME
+hockeyAppId=FIXME
+googleAnalyticsId=FIXME
+facebookAppId=FIXME
```
Comment (roundhill) [inline Simplenote/gradle.properties-example]: We should update the `README` to rename this file maybe? Doesn't need to be in this PR

Comment (roundhill) [inline Simplenote/gradle.properties-example]: Also, we can kill `facebookAppId` and `hockeyAppId`

**Comment by maxme** [comment]
@roundhill I updated the gradle.properties and README files.

**Comment by roundhill** [comment]
:shipit:!

**Pull request merged**